src/com/chessBOTP/Main.java

```diff
@@ -10,179 +12,225 @@
+                } else {
+                    BKFirstMove = false;
+                }
+
+            // Store chosen cell piece properties for undo purposes
             Cells cell = new Cells(chosenCell.CONTAINS, chosenCell.pieceColor, chosenCell.piece);
             turnHandler.getCurrentPlayer().addMove(cell);
 
-            //**Move the Piece**//
+            // Move the clicked piece to the chosen cell
             chosenCell.CONTAINS = prevChosenCell.CONTAINS; //The previous cell will now move to the new cell
-            chosenCell.pieceColor = prevChosenCell.pieceColor; //The newly clicked cell will contain the color of the previous cell
-            chosenCell.setIcon(prevChosenCell.getIcon()); //The newly clicked cell will contain the text of the previous cell
+            chosenCell.pieceColor = prevChosenCell.pieceColor; //The newly clicked cell will contain the color property of the previous cell
+            chosenCell.setIcon(prevChosenCell.getIcon()); //The newly clicked cell will contain the icon of the previous cell
             chosenCell.piece = chosenCell.getIcon();
+
+            // Store the properties of the previously chosen cell for undo purposes
             turnHandler.getCurrentPlayer().addMove(chosenCell); //Add the move to the current player's moves
 
-            //**Reset The previous Cell**//
+            // Reset the previously clicked cell
             prevChosenCell.CONTAINS = 0;
             prevChosenCell.pieceColor = 0;
             prevChosenCell.setIcon(null);
             prevChosenCell.piece = prevChosenCell.getIcon();
             turnHandler.getCurrentPlayer().addMove(prevChosenCell);
+
+            // Calculate future moves if they result to a check
+            isSuggesting = false;
+            calculateFutureMove(chessboard.getCells(), isSuggesting);
+
             turnHandler.nextTurn(); //Change the turn to the next player
+            check();
 
             allowedToMove = false;
-            resetColAvailCells(chessboard.getCells());
+            resetAvailCells(chessboard.getCells());
             return;
         }
 
-        if (turnHandler.getCurrentPlayer().getPlayerColor() == chosenCell.pieceColor) {
+        if(turnHandler.getCurrentPlayer().getPlayerColor() == chosenCell.pieceColor) {
             chosenCell.setBackground(Color.YELLOW);
-            setColAvailCells(chosenCell, turnHandler.getCurrentPlayer().getPlayerColor());
+            isSuggesting = true;
+            suggestAvailCells(chosenCell, turnHandler.getCurrentPlayer().getPlayerColor(), isSuggesting);
+            prevChosenCell = chosenCell;
+            allowedToMove = true;
         }
     }
 
-    public static Cells calculateFutureMove(Cells chosenCell, int currentColorPiece, int i, int[][]moves) {
+    public static Cells calculateCurrentMove(Cells chosenCell, int currentColorPiece, int i, int[][]moves) {
         int x = chosenCell.posX + (currentColorPiece * moves[i][1]);
```

Comment:
change the code to this:
int x = chosenCell.posX + moves[i][0];

to suit the format changes of cells[x][y]